EIPS/eip-6093.md

```diff
@@ -0,0 +1,376 @@
+
+This EIP can not be enforced on non-upgradeable already deployed tokens, however, these tokens generally use similar conventions with small variations such as:
+
+- including/removing the [domain](#domain).
+- using different [error prefixes](#error-prefixes).
+- including similar [subjects](#actions-and-subjects).
+- changing the grammar order.
+
+Upgradeable contracts MAY be upgraded to implement this EIP.
+
+Implementers and DApp developers that implement special support for tokens that are compliant with this EIP, SHOULD tolerate different errors emitted by non-compliant contracts, as well as classic revert strings.
+
+## Reference Implementation
+
+### Solidity
+
+```solidity
+pragma solidity ^0.8.4;
+
+/// @title Standard ERC20 Errors
+/// @dev See https://eips.ethereum.org/EIPS/eip-20
+///  https://eips.ethereum.org/EIPS/eip-6093
+interface ERC20Errors {
+    error ERC20InsufficientBalance(address sender, uint256 balance, uint256 needed);
+    error ERC20InvalidSender(address sender);
+    error ERC20InvalidReceiver(address receiver);
+    error ERC20InsufficientAllowance(address spender, uint256 allowance, uint256 needed);
+    error ERC20InvalidApprover(address approver);
+    error ERC20InvalidSpender(address spender);
+}
+
+/// @title Standard ERC721 Errors
+/// @dev See https://eips.ethereum.org/EIPS/eip-721
+///  https://eips.ethereum.org/EIPS/eip-6093
+interface ERC721Errors {
+    error ERC721InvalidOwner(address sender, uint256 tokenId, address owner);
+    error ERC721InvalidSender(address sender);
+    error ERC721InvalidReceiver(address receiver);
+    error ERC721InsufficientApproval(address operator, uint256 tokenId);
+    error ERC721InvalidApprover(address approver);
+    error ERC721InvalidOperator(address operator);
+}
+
+/// @title Standard ERC1155 Errors
+/// @dev See https://eips.ethereum.org/EIPS/eip-1155
+///  https://eips.ethereum.org/EIPS/eip-6093
+interface ERC1155Errors {
+    error ERC1155InsufficientBalance(address sender, uint256 balance, uint256 needed, uint256 tokenId);
+    error ERC1155InvalidSender(address sender);
+    error ERC1155InvalidReceiver(address receiver);
+    error ERC1155InsufficientApproval(address operator, uint256 tokenId);
+    error ERC1155InvalidApprover(address approver);
+    error ERC1155InvalidOperator(address operator);
+    error ERC1155InvalidArrayLength(uint256 idsLength, uint256 valuesLength);
+}
+```
+
+## Security Considerations
+
+There are no known signature hash collision for the specified errors.
+
+Tokens upgraded to implement this EIP may break assumptions in other systems relying on revert strings.
+
+## Copyright
+
+Copyright and related rights waived via [CC0](../LICENSE.md).
+
+## Citation
+
+Please cite this document as:
+
+Ernesto García, Francisco Giordano, Hadrien Croubois, "EIP-6093: Standard Token Errors [DRAFT]," Ethereum Improvement Proposals, no. 6093, November 2022. [Online serial]. Available: https://eips.ethereum.org/EIPS/eip-6093.
```

Comment:
```suggestion
```
This section is automatically generated by Jekyll.

Reply:
Amazing, thanks 🙌